xcode/Subconscious/Shared/Components/Common/Byline/PetnameView.swift

```diff
@@ -7,54 +7,122 @@
 
 import SwiftUI
 
+public enum NameVariant {
+    case petname(Slashlink, Petname.Name)
+    case proposedName(Slashlink, Petname.Name)
+    case selfNickname(Slashlink, Petname.Name)
+}
+
+extension UserProfile {
+    func toNameVariant() -> NameVariant? {
+        switch (
+            self.category,
+            self.ourFollowStatus,
+            self.nickname,
+            self.address.petname?.leaf
+        ) {
```

Comment:
Couldn't find a specific recommendation in the google swift styleguide for formatting long switch tuples like this, so this is a best guess.

Reply:
Works for me